pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -481,6 +482,11 @@ public <T> ConsumerBuilder<T> newConsumerBuilder(Schema<T> schema) throws Pulsar
         return this.client.newConsumer(schema);
     }
 
+    @Override
+    public <X> FunctionRecord.FunctionRecordBuilder<X> newOutputRecordBuilder() {
```

Comment:
Instead of adding a new API here, do you think about creating a util class with a record generation method?

Reply:
I think it is better to not add a general purpose API to build "Record" instances.
A Record is like a Message and you cannot build a Message using the client API or a utility class.
The record may be tied to some internal context or pre-configured (as it does currently in this PR).

So it is better to use Context as a starting point for building a new record, the same way it happens with newOutputMessage()